design/one-pager-generic-usage-type.md

```diff
@@ -0,0 +1,397 @@
+(i.e. `spec.of` as a list) as well, considering it provides more flexibility
+without introducing additional complexity to the implementation.
+
+### Implementation
+
+The implementation of this feature will be done in two parts:
+
+1. The `Usage` type and its controller, which will manage the lifecycle of the
+   resource.
+2. The admission webhook that will enforce the usage relationships.
+
+Both will be implemented in the Crossplane core repository and would not require
+any changes in the providers.
+
+#### Usage Controller
+
+The usage controller will be responsible for managing the lifecycle of the usage
+relationship by:
+
+- Resolving selectors, if any.
+- Adding owner reference from the using resource (i.e., `spec.by`) to the `Usage`.
+- Preventing deletion of `Usage` before the using resource is deleted.
+- Adding or removing admission webhook rules to intercept the `DELETE` requests
+  for the used resource(s).
+
+**Resolving selectors:**
+
+The API will support defining selectors for both the using and used resources.
+Initially, we will only support resolving references with the following rules:
+
+- Resolution will be made once when the `Usage` is created.
+- If multiple resources match the selector, a random one will be
+  selected.
+
+The [existing policies] for resolving Managed Resource references does not
+necessarily apply to the `Usage`. For example, supporting an `optional`
+reference while defining a usage relationship does not make sense. Also, the
+[reasoning behind] supporting different policies for MRs does not apply to
+the `Usage`. However, depending on the feedback, we may consider supporting
+resolve policies to configure whether the resolution should be made once or
+continuously in a future iteration.
+
+**Adding owner reference:**
+
+We want the `Usage` to be deleted when the using resource is deleted. This will
+be achieved by adding an owner reference from the using resource to the `Usage`.
+
+**Preventing early deletion of the `Usage`:**
+
+Typically, we expect a `Usage` resource to be defined as part of the same
+composition as the using and used resources. This means the `Usage` will get
+the delete request when the composite resource is deleted. In this case, the
+controller will prevent the deletion of the `Usage` until the using resource is
+deleted with the help of a finalizer.
+
+**Defining admission webhook rules:**
+
+The admission webhook will intercept the `DELETE` requests for the used
+resource(s). To achieve this, the controller will add [webhook rules] for the
+used resource(s) when a `Usage` is created. It is also the controller's job to
+remove these rules when the `Usage` is deleted, and no other `Usage` resources
+reference the same used resource(s).
+
+#### Admission Webhook
+
+The admission webhook will be responsible for intercepting the `DELETE` requests
+for the used resource(s) and rejecting them with a ["409 Conflict" error code].
```

Comment:
Would we fail open or closed if the webhook failed to process the request in time?

https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#failure-policy

Reply:
We probably have to fail closed because otherwise the 409 is ignored.

Reply:
xref https://github.com/crossplane/crossplane/pull/4215/files#r1236446471

Reply:
Agree with @sttts. 

I don't see a reason to fail open here, so we will set `failurePolicy: Ignore`.